pkg/controller/instance/instance_controller.go

```diff
@@ -231,12 +232,12 @@ func (r *Reconciler) handleError(err error, instance *kudov1alpha1.Instance) err
 	}
 
 	// determine if retry is necessary based on the error type
-	if exErr, ok := err.(*executionError); ok {
-		if exErr.eventName != nil {
-			r.Recorder.Event(instance, "Warning", kudo.StringValue(exErr.eventName), err.Error())
+	if exErr, ok := err.(*ExecutionError); ok {
```

Comment:
I was thinking of getting rid of it altogether in a follow-up PR. It became somewhat unnecessary because the only error we actually return from the plan execution are fatal. On the transient errors, we simply move on to the next phase/step.